exec/java-exec/src/main/java/org/apache/drill/exec/server/rest/DrillRestServer.java

```diff
@@ -230,6 +230,27 @@ public WebUserConnection provide() {
     public void dispose(WebUserConnection instance) {
 
     }
+
+    /**
+     * Creates session user principal. If impersonation is enabled without authentication and User-Name header is present and valid,
+     * will create session user principal with provided user name, otherwise anonymous user name will be used.
+     * In both cases session user principal will have admin rights.
+     *
+     * @param config drill config
+     * @param request client request
+     * @return session user principal
+     */
+    private Principal createSessionUserPrincipal(DrillConfig config, HttpServletRequest request) {
+      final boolean checkForUserName = !config.getBoolean(ExecConstants.USER_AUTHENTICATION_ENABLED) && config.getBoolean(ExecConstants.IMPERSONATION_ENABLED);
+      if (checkForUserName) {
```

Comment:
Done.

Reply:
We should not require `User-Name` for each request since it is needed only when we need to query the data. I will leave this method but also add `UserNameFilter` that would be applied to all `/query` and `/query.json` requests and check if header is provided, i.e. we would allow using `anonymous` user if `User-Name` is not provided since we'll know that `UserNameFilter` will filter out request where `User-Name` is required but not set.

Reply:
Makes sense. Thank You for explanation